src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2988,11 +3053,16 @@ private void setResponseClosed()
 
                 if (Http2State.closed(state))
                 {
-                    streams.remove(streamId);
-                    streamsActive[streamId & 0x01]--;
+                    removeStream();
                 }
             }
 
+            private void removeStream()
+            {
+                streams.remove(streamId);
+                streamsActive[streamId & 0x01]--;
+            }
```

Comment:
This should be `removeStreamIfNecessary` and include the check for `if (Http2State.closed(state))`
The `else if (streams.containsKey(streamId))` above needs review to make sure the `state` is accurate for the request.